src/shared/utilities/vsCodeUtils.ts

```diff
@@ -0,0 +1,111 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
+export const localize = nls.loadMessageBundle()
+
+import { ErrorOrString, getLogger, Logger } from '../logger'
+
+// ------- Experimental combined output channel & traditional logger  --------------
+
+type LogLevel = 'debug' | 'info' | 'warn' | 'error'
+
+interface LogParams {
+    args: ErrorOrString[],
+    channel: vscode.OutputChannel,
+    level: LogLevel,
+    logger: Logger,
+    msg: string,
+    nlsKey: string,
+}
+
+// TODO: Leverage 3rd party code: https://raw.githubusercontent.com/Microsoft/vscode-nls/master/src/main.ts
+function defaultTranslation(message: string, ...args: any[]): string {
+    if (!args.length) {
+        return message
+    }
+
+    return `${message}, args: ${JSON.stringify(args)}`
```

Comment:
Please try-catch this and fail gracefully (return message?), in case stringify were to raise an exception (eg: circular reference or something)

Reply:
✔️